### DISCUSSION TITLE
What type should be used to return an URL?

### DISCUSSION THREAD

**Discussion by Glitchy-Tozier** [rwf2/Rocket#2224] [Questions]
The simplest way might be to return a `String`.
```rs
fn handler() -> String {
    ...
    url
}
```
That `String` could be created manually, or it could be created using `uri!().to_string()`.

However, I'm wondering whether there is a more appropriate return-type for those kinds of situation.

---

Some context: The response is used to tell the client where to navigate next, after the current POST-request. I was told that Redirect would not be the correct call.

**Comment by the10thWiz** [comment]
I believe the `Url` type implements Responder, but otherwise it's probably best to not to return a Url like this.

Why isn't Redirect the correct call? At first glance, Redirect makes the most sense - the browser will automatically fetch the next page, and start loading it. However, if there is additional work on the client side, or it's a single page application, a redirect may not be the best option. However, if that is the case, the Url should be wrapped into some data structure, such as a JSON or XML object. The `Url` type does support both `Serialize` & `Deserialize`, so this should require very little effort.

**Reply by Glitchy-Tozier** [reply]
> Why isn't Redirect the correct call? At first glance, Redirect makes the most sense - the browser will automatically fetch the next page, and start loading it.

Thank you for your response!
I liked your idea and tried it out, it didn't work, however.

The issue (I think) comes from this handler accepting POST-requests from a form. Redirecting _does indeed happen_, but that only means the POST-request will be sent somewhere else, still receiving the new data (the HTML of the new page), but not navigating to any new page.
